src/plugins/pillbox-menus/components/Menu.tsx

```diff
@@ -193,6 +198,95 @@ export default class Menu extends Component<{
   }
 }
 
+let counter = 0;
+function colorListOption(
+  pm: PillboxMenus,
+  item: ConfigItem,
+  plugin: SpecificPlugin,
+  settings: GenericSettings
+) {
+  // const clItem = item as ConfigItemColorList;
+
+  const setValue = (newValue: string[]) =>
+    pm.expandedPlugin &&
+    pm.dsm.setPluginSetting(pm.expandedPlugin, item.key, newValue);
+
+  const value = () => settings[item.key] as string[];
+
+  return (
+    <div class="dsm-settings-item">
+      <div class="dsm-settings-color-list-container">
+        <Tooltip tooltip={configItemDesc(plugin, item)} gravity="n">
+          <label for={`dsm-settings-item__input-${item.key}`}>
+            {configItemName(plugin, item)}
+          </label>
+        </Tooltip>
+        {IfElse(() => value().length > 0, {
+          true: () => (
+            <div class="flex">
+              <For
+                each={() =>
+                  (settings[item.key] as string[]).map((e, i) => [e, i])
+                }
+                // TODO: find a better way of dealing with this
+                key={() => counter++}
```

Comment:
This can probably be `key={([e,i]: [string, number]) => i}`.
If that doesn't update on change, then I suggest using instead `<For each={() => (settings[item.key] as string[]).map((e, i) => i)}>`, then Inside the For, replace `value={v}` with `value={() => settings[item.key][i]}`

Reply:
As for your first suggestion, that doesn't seem to work correctly as far as I can tell--- all appends and deletions seem to happen to the end of the list instead of at the intended location.

The second one would probably work if not for the fact that in theory someone could put duplicate colors.

Reply:
I might be able to get away with making some sort of a "hybrid key" made of both the index and the hex code. Ideally each element would have its own key that'd "stick around" with it as it moved through the collection, though that'd make things rather annoying to implement unless I decided to expose that functionality of "keyed arrays" directly to the plugins.